cmd/poktrolld/cmd/migrate/types.go

```diff
@@ -0,0 +1,167 @@
+	// It includes a slice of MorseAccount objects which are populated, by transforming
+	// the input MorseStateExport into the output MorseAccountState.
+	accountState *migrationtypes.MorseAccountState
+
+	// lastAccTotalBalance is the most recently accumulated balances of all Morse
+	// accounts which have been processed.
+	lastAccTotalBalance cosmosmath.Int
+	// lastAccTotalAppStake is the most recently accumulated application stakes of
+	// all Morse accounts which have been processed.
+	lastAccTotalAppStake cosmosmath.Int
+	// lastAccTotalSupplierStake is the most recently accumulated supplier stakes of
+	// all Morse accounts which have been processed.
+	lastAccTotalSupplierStake cosmosmath.Int
+	// numAccounts is the number of accounts that have been processed.
+	numAccounts uint64
+	// numApplications is the number of applications that have been processed.
+	numApplications uint64
+	// numSuppliers is the number of suppliers that have been processed.
+	numSuppliers uint64
+}
+
+// nextIdx returns the next index to be used when appending a new account to the accounts slice.
+func (miw *morseImportWorkspace) nextIdx() uint64 {
+	return uint64(len(miw.accountState.Accounts))
+}
+
+// hasAccount returns true if the given address is present in the accounts slice.
+func (miw *morseImportWorkspace) hasAccount(addr string) bool {
+	_, ok := miw.addressToIdx[addr]
+	return ok
+}
+
+// debugLogProgress prints debug level logs indicating the progress of the import,
+// according to flagDebugAccountsPerLog.
+func (miw *morseImportWorkspace) debugLogProgress(accountIdx int) {
+	logger.Debug().
+		Int("account_idx", accountIdx).
+		Uint64("num_accounts", miw.numAccounts).
+		Uint64("num_applications", miw.numApplications).
+		Uint64("num_suppliers", miw.numSuppliers).
+		Str("total_balance", miw.lastAccTotalBalance.String()).
+		Str("total_app_stake", miw.lastAccTotalAppStake.String()).
+		Str("total_supplier_stake", miw.lastAccTotalSupplierStake.String()).
+		Str("grand_total", miw.lastAccGrandTotal().String()).
+		Msg("processing accounts...")
+}
+
+// infoLogComplete prints info level logs indicating the completion of the import.
+func (miw *morseImportWorkspace) infoLogComplete() error {
+	accountStateHash, err := miw.accountState.GetHash()
+	if err != nil {
+		return err
+	}
+
+	logger.Info().
+		Uint64("num_accounts", miw.numAccounts).
+		Uint64("num_applications", miw.numApplications).
+		Uint64("num_suppliers", miw.numSuppliers).
+		Str("total_balance", miw.lastAccTotalBalance.String()).
+		Str("total_app_stake", miw.lastAccTotalAppStake.String()).
+		Str("total_supplier_stake", miw.lastAccTotalSupplierStake.String()).
+		Str("grand_total", miw.lastAccGrandTotal().String()).
+		Str("morse_account_state_hash", fmt.Sprintf("%x", accountStateHash)).
+		Msg("processing accounts complete")
+	return nil
+}
+
+// lastAccGrandTotal returns the sum of the lastAccTotalBalance, lastAccTotalAppStake,
```

Comment:
Mentioned this elsewhere, but I'm tripped up by the word `lastAcc`. 

Is it just `currentCumulative`?